plaidml/edsl/tests/edsl_test.py

```diff
@@ -518,6 +518,40 @@ def test_skip(self):
 }
 '''
 
+    def test_cast_scalar(self):
+        I = Placeholder(plaidml.DType.INT32, {})
+        O = cast(I, plaidml.DType.FLOAT32)
+        program = Program('cast', [O])
+        # CHECK-LABEL: test_cast_scalar
+        expected = '''
+module {
+  func @cast(%arg0: tensor<si32>) -> tensor<f32> {
+    %0 = "eltwise.cast"(%arg0) : (tensor<si32>) -> tensor<f32>
+    return %0 : tensor<f32>
+  }
+}
+'''
+        self.checkProgram(program, [
+            (I, np.array([(3)])),
+        ], [3.0])
```

Comment:
This is where using c++ would be a better test, because python might automatically cast for you, hiding the actual data type.

Reply:
Good point. I'll switch both tests over to c++.